### ISSUE TITLE
Formatting

### ISSUE THREAD

**Issue by PaulXiCao** [CGAL/cgal#5985]
I am a new user of CGAL and want to contribute to the project. Currently I am working through documentation and examples. I came across some inconsistencies and want to ask if such pull request are welcome and have a chance of merging.

## Formatting inconsistencies
Would it be possible to enforce a formatting style guide for new code, e.g. via `clang-format`?

### White spaces in function arguments
Function arguments are sometimes seperated by a white space and sometimes not.
Example file: [Kernel_23/examples/Kernel_23/points_and_segment.cpp](https://github.com/CGAL/cgal/blob/master/Kernel_23/examples/Kernel_23/points_and_segment.cpp#L18).
```cpp
...
  Segment_2 s(p,q); // line 18
  Point_2 m(5, 9);
...
```

### White spaces in template arguments
Sometimes there are white spaces inside of the template argument list, sometimes not.
Example file: [Kernel_23/include/CGAL/Aff_transformation_2.h](https://github.com/CGAL/cgal/blob/master/Kernel_23/include/CGAL/Aff_transformation_2.h)
```cpp
...
template <class R_>  // line 25
...
template < class R >  // line 104
...
```

## Newline character
`\n` is preferred over `std::endl` if flushing of buffer is unneeded.
Example file: [Kernel_23/examples/Kernel_23/points_and_segment.cpp](https://github.com/CGAL/cgal/blob/master/Kernel_23/examples/Kernel_23/points_and_segment.cpp).

**Comment by sloriot** [comment]
While contributions are welcome, we tend to avoid automatic formatting as it can create conflicts with open pull requests and also make it hard to follow changes in the history of a file.
